packages/core/src/storage/vectorStore/TiDBVectorStore.ts

```diff
@@ -0,0 +1,322 @@
+import type mysql from "mysql2/promise";
+import type { PoolOptions, RowDataPacket } from "mysql2/promise";
+
+import type {
+  VectorStore,
+  VectorStoreQuery,
+  VectorStoreQueryResult,
+} from "./types.js";
+
+import type { GenericFileSystem } from "@llamaindex/env";
+import type { BaseNode, Metadata } from "../../Node.js";
+import { Document, MetadataMode } from "../../Node.js";
+
+export const TIDB_VECTOR_TABLE = "llamaindex_embedding";
+
+interface DocumentEmbedding extends RowDataPacket {
+  id: string;
+  document: string;
+  metadata: Metadata;
+  embeddings: number[];
+  score: number;
+}
+
+/**
+ * Provides support for writing and querying vector data in TiDB.
+ */
+export class TiDBVectorStore implements VectorStore {
```

Comment:
did you check if the data generated is compatible with the python version (see https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/vector_stores/llama-index-vector-stores-tidbvector/llama_index/vector_stores/tidbvector/base.py) ?

Reply:
grest, I will make some adjustments to be consistent with the Python version.